lib/ecto/repo.ex

```diff
@@ -171,7 +171,11 @@ defmodule Ecto.Repo do
         end, ecto_conn_pid: entry.connection_pid)
       end
 
-      defoverridable [log: 1, __pool__: 0]
+      def after_connect(conn) do
```

Comment:
Just don't define this function so the `function_exported?` check works (and pls remove the defcallback).